src/goto-symex/symex_decl.cpp

```diff
@@ -67,6 +67,21 @@ void goto_symext::symex_decl(statet &state, const symbol_exprt &expr)
   std::size_t generation = state.increase_generation(l1_identifier, ssa);
   CHECK_RETURN(generation == 1);
 
+  const exprt fields = state.field_sensitivity.get_fields(ns, state, ssa);
+  if(fields != ssa)
+  {
+    std::set<symbol_exprt> fields_set;
+    find_symbols(fields, fields_set);
+
+    for(const auto &l1_symbol : fields_set)
+    {
+      ssa_exprt field_ssa = to_ssa_expr(l1_symbol);
+      std::size_t field_generation =
+        state.increase_generation(l1_symbol.get_identifier(), field_ssa);
```

Comment:
This is surely either-or -- if we're doing this we should not create the non-canonical symbol `l1_identifier`. So I'd say remove the `if(fields != ssa)` and the `increase_generation` call directly above it.

Reply:
Done.